src/backends/gstreamer/audio_sink.rs

```diff
@@ -87,9 +61,13 @@ impl AudioSink for GStreamerAudioSink {
                     .into();
                 buffer.set_pts(pts);
                 buffer.set_duration(next_pts - pts);
-                let mut map = buffer.map_writable().unwrap();
-                let data = map.as_mut_slice();
-                graph_.process(data, rate);
+                
+                let mut chunks = graph_.process(rate);
+                debug_assert!(chunks.len() == 1);
+                let data = &mut chunks.blocks[0].data;
+                let data = data.as_mut_byte_slice().expect("casting failed");
+                buffer.copy_from_slice(0, data).expect("copying failed");
```

Comment:
This looks like a potentially unneeded copy. You can wrap a buffer around any `AsRef<[u8]>` or `AsMut<[u8]>` if that helps anything here

Reply:
How would I do that?

Reply:
Currently there is only minimal support for custom memory types in the bindings (see [here](https://github.com/sdroege/gstreamer-rs/issues/12)), but it's probably sufficient for what you need. [From a mutable slice](https://github.com/sdroege/gstreamer-rs/blob/1192a027b40c5ad46cd998cfa7fd11bc527bd1a9/gstreamer/src/buffer.rs#L75-L95) and [from an immutable slice](https://github.com/sdroege/gstreamer-rs/blob/1192a027b40c5ad46cd998cfa7fd11bc527bd1a9/gstreamer/src/buffer.rs#L97-L117)

Reply:
Ah. Due to the way rustdoc deals with typedefs those functions don't show up in the docs where you'd expect 😄

Reply:
Yeah, personally I'm not using rustdoc because of that for anything built on top of gtk-rs. Easier to navigate the sources :)

Reply:
Okay so it looks like I'd need to convert between `Box<[u8]>` and `Box<[f32]>` here (and in future versions, between vecs), and byte-slice-cast doesn't support that yet. Filing a bug at https://github.com/sdroege/byte-slice-cast/issues/1 . It's not hard to write but I don't want to do one-off unsafe code in servo-media for it.